google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITPitrUpdateDatabaseTest.java

```diff
@@ -0,0 +1,195 @@
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+
+@Category(ParallelIntegrationTest.class)
+@RunWith(JUnit4.class)
+public class ITPitrUpdateDatabaseTest {
+
+  private static final Duration OPERATION_TIMEOUT = Duration.ofMinutes(2);
+  private static final String VERSION_RETENTION_PERIOD = "7d";
+
+  @ClassRule public static IntegrationTestEnv env = new IntegrationTestEnv();
+  private static DatabaseAdminClient dbAdminClient;
+  private static DatabaseClient dbClient;
+  private static String instanceId;
+  private static String databaseId;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    assumeFalse("PITR-lite features are not supported by the emulator", isUsingEmulator());
+
+    final RemoteSpannerHelper testHelper = env.getTestHelper();
+    final String projectId = testHelper.getOptions().getProjectId();
+    instanceId = testHelper.getInstanceId().getInstance();
+    databaseId = testHelper.getUniqueDatabaseId();
+    dbAdminClient = testHelper.getClient().getDatabaseAdminClient();
+
+    createDatabase(dbAdminClient, instanceId, databaseId, Collections.<String>emptyList());
+    updateVersionRetentionPeriod(dbAdminClient, instanceId, databaseId, VERSION_RETENTION_PERIOD);
+
+    dbClient =
+        testHelper.getClient().getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    if (!isUsingEmulator()) {
+      dbAdminClient.dropDatabase(instanceId, databaseId);
+    }
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughGetDatabase() {
+    final Database database = dbAdminClient.getDatabase(instanceId, databaseId);
+
+    assertThat(database.getVersionRetentionPeriod()).isEqualTo(VERSION_RETENTION_PERIOD);
+    assertThat(database.getEarliestVersionTime()).isNotNull();
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughListDatabases() {
+    final Page<Database> page = dbAdminClient.listDatabases(instanceId);
+
+    for (Database database : page.iterateAll()) {
+      if (!database.getId().getDatabase().equals(databaseId)) {
+        continue;
+      }
+      assertThat(database.getVersionRetentionPeriod()).isEqualTo(VERSION_RETENTION_PERIOD);
+      assertThat(database.getEarliestVersionTime()).isNotNull();
+    }
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughGetDatabaseDdl() {
+    final List<String> ddls = dbAdminClient.getDatabaseDdl(instanceId, databaseId);
+
+    boolean hasVersionRetentionPeriodStatement = false;
+    for (String ddl : ddls) {
+      hasVersionRetentionPeriodStatement =
+          ddl.contains("version_retention_period = '" + VERSION_RETENTION_PERIOD + "'");
+      if (hasVersionRetentionPeriodStatement) {
+        break;
+      }
+    }
+    assertThat(hasVersionRetentionPeriodStatement).isTrue();
+  }
+
+  @Test
+  public void returnsTheVersionRetentionPeriodSetThroughInformationSchema() {
+    final ResultSet rs =
```

Comment:
This query will list the `OPTION_VALUE` of all databases on the test instance. If this test is executed against an existing instance with multiple databases having a `version_retention_period`, it could fail.

Reply:
I am wondering how could this happen, since I am using the `dbClient` for which we need to specify a database to create. Or maybe I am missing something?

Reply:
Sorry, you're completely right. I forgot that the INFORMATION_SCHEMA is always database-specific.